packages/cli/src/commands/validator/status.ts

```diff
@@ -30,82 +55,72 @@ export default class ValidatorStatus extends BaseCommand {
 
   static examples = [
     'status --validator 0x5409ED021D9299bf6814279A6A1411A7e866A631',
-    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c',
-    'status --signer 0x738337030fAeb1E805253228881d844b5332fB4c --lookback 100',
+    'status --all --lookback 100',
   ]
 
   requireSynced = true
 
+  async getStatus(
+    signer: Address,
+    blocks: Block[],
+    electedSigners: Address[],
+    frontRunnerSigners: Address[]
+  ): Promise<ValidatorStatusEntry> {
+    const accounts = await this.kit.contracts.getAccounts()
+    const validator = await accounts.signerToAccount(signer)
+    const name = (await accounts.getName(validator)) || ''
+    const electedIndex = electedSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const frontRunnerIndex = frontRunnerSigners.map((a) => eqAddress(a, signer)).indexOf(true)
+    const proposedCount = blocks.filter((b) => b.miner === signer).length
+    let signedCount = 0
+    if (electedIndex >= 0) {
+      signedCount = blocks.filter((b) =>
+        bitIsSet(parseBlockExtraData(b.extraData).parentAggregatedSeal.bitmap, electedIndex)
+      ).length
+    }
+    return {
+      name,
+      address: validator,
+      signer,
+      elected: electedIndex >= 0,
+      frontRunner: frontRunnerIndex >= 0,
+      proposed: proposedCount,
+      signatures: signedCount / blocks.length,
+    }
+  }
+
   async run() {
     const res = this.parse(ValidatorStatus)
-
-    // Check that the specified validator or signer meets the validator requirements.
-    const checker = newCheckBuilder(this, res.flags.signer)
-    if (res.flags.validator) {
-      const account = res.flags.validator
-      checker
-        .isAccount(account)
-        .isValidator(account)
-        .meetsValidatorBalanceRequirements(account)
-    } else if (res.flags.signer) {
-      checker
-        .isSignerOrAccount()
-        .signerMeetsValidatorBalanceRequirements()
-        .signerAccountIsValidator()
+    const accounts = await this.kit.contracts.getAccounts()
+    const validators = await this.kit.contracts.getValidators()
+    let signers: string[] = []
+    const checker = newCheckBuilder(this)
+    if (res.flags.signer) {
+      signers = [res.flags.signer]
+      checker.isAccount(res.flags.validator).isValidator(res.flags.validator)
+      await checker.runChecks()
+    } else if (res.flags.validator) {
+      const validator = await accounts.signerToAccount(res.flags.signer)
+      signers = [validator]
+      checker.isAccount(validator).isValidator(validator)
+      await checker.runChecks()
```

Comment:
I think this conditional might be switched because `res.flags.validator` is used when only `res.flags.signer` is defined and vice versa